src/auto/cell_area.rs

```diff
@@ -237,13 +237,37 @@ impl<O: IsA<CellArea>> CellAreaExt for O {
         }
     }
 
-    //fn foreach<P: Into<Option</*Unimplemented*/Fundamental: Pointer>>>(&self, callback: /*Unknown conversion*//*Unimplemented*/CellCallback, callback_data: P) {
-    //    unsafe { TODO: call ffi::gtk_cell_area_foreach() }
-    //}
+    fn foreach<P: FnMut(&CellRenderer) -> bool>(&self, callback: P) {
+        let callback_data: Option<P> = callback.into();
+        unsafe extern "C" fn callback_func<P: FnMut(&CellRenderer) -> bool>(renderer: *mut ffi::GtkCellRenderer, data: glib_ffi::gpointer) -> glib_ffi::gboolean {
+            let renderer = from_glib_borrow(renderer);
+            let callback: *mut P = data as *const _ as usize as *mut P;
```

Comment:
This is wrong. The `user_data` is a `&Option<P>`, i.e. `*mut Option<P>` or whatever. You dropped the `Option<_>` part here.

Reply:
Also unclear why it's an `Option<_>` at all as the function is not optional :)

Reply:
It's not anymore. :)